src/smolagents/models.py

```diff
@@ -300,8 +300,8 @@ def _prepare_completion_kwargs(
 
     def get_token_counts(self) -> Dict[str, int]:
         return {
-            "input_token_count": self.last_input_token_count,
-            "output_token_count": self.last_output_token_count,
+            "input_token_count": int(self.last_input_token_count or 0),
```

Comment:
This seems to be a lossy conversion to int by defaulting to 0. The original default value is `None`. I'd do instead `dict[str, int | None]` (note: this is a supported PEP484 syntax in Python 3.10).

Reply:
`None` signifies that the model hasn't made any calls. But I'm not sure what the value would be if the model call returns an erroneous count.

Reply:
Oki, I see the None has a value here signifies no calls. I'll update it accordingly, maybe adding docstring explaining this behavior.